src/main/java/org/embulk/util/config/Version.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2020 The Embulk project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.embulk.util.config;
+
+import java.net.URL;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+
+/**
+ * A utility class defining a constant of Jackson's Module version.
+ */
+final class Version {
+    static final com.fasterxml.jackson.core.Version JACKSON_MODULE_VERSION = getVersion();
+
+    private Version() {
+        // No instantiation.
+    }
+
+    static com.fasterxml.jackson.core.Version parseVersionForTesting(final String version) {
+        return parseVersion(version);
+    }
+
+    private static com.fasterxml.jackson.core.Version getVersion() {
+        try {
+            return parseVersion(getImplementationVersion());
+        } catch (final Throwable any) {  // It must not fail.
+            return com.fasterxml.jackson.core.Version.unknownVersion();
+        }
+    }
+
+    private static com.fasterxml.jackson.core.Version parseVersion(final String implementationVersion) {
+        final String[] versionAndSnapshotInfo = implementationVersion.split("-", 2);
+        final String version;
+        final String snapshotInfo;
+        if (versionAndSnapshotInfo.length == 1) {
+            version = versionAndSnapshotInfo[0];
+            snapshotInfo = null;
+        } else if (versionAndSnapshotInfo.length == 2) {
+            version = versionAndSnapshotInfo[0];
+            snapshotInfo = versionAndSnapshotInfo[1];
+        } else {
+            throw new IllegalArgumentException("Empty version string.");
+        }
+
+        final String[] versionSplit = version.split("\\.");
```

Comment:
All the subsequent use of `versionSplit` better to trim() to avoid NFE

Reply:
I intentionally skipped such good error handling here for the same reason above.

* In case any unexpected errors, it should go `unknownVersion` without any complicated error handling and recovery.
* If such `versionSplit` is needed for a version (specified in JAR Manifest from `build.gradle`), that should be caught by `TestVersion.java` before releasing.

Reply:
got it